### PR TITLE
Added methods to use struqture 2.0

### PR DESCRIPTION
**Pull request by kbarkhqs**

### CODE REVIEW

File: struqture-py/src/fermions/fermionic_hamiltonian_system.rs [hunk 1]
```diff
@@ -130,4 +131,64 @@ impl FermionHamiltonianSystemWrapper {
...
+    ///     FermionHamiltonianSystem: The struqture 1.x FermionHamiltonianSystem created from the struqture 2.x FermionHamiltonian.
+    ///
+    /// Raises:
+    ///     TypeError: If the input is not a struqture 2.x FermionHamiltonian.
+    ///     ValueError: Conversion failed.
+    #[staticmethod]
+    pub fn from_struqture_2(input: &Bound<PyAny>) -> PyResult<FermionHamiltonianSystemWrapper> {
+        Python::with_gil(|_| -> PyResult<FermionHamiltonianSystemWrapper> {
+            let source_serialisation_meta = input.call_method0("_get_serialisation_meta").map_err(|_| {
+                PyTypeError::new_err("Trying to use Python object as a struqture-py object that does not behave as struqture-py object. Are you sure you have the right type to all functions?".to_string())
```
Comment (rreiner-hqs) [inline struqture-py/src/fermions/fermionic_hamiltonian_system.rs:150]: I would suggest saving error strings into a variable for reusability, and also maybe establishing a standard line length that we want to keep

File: struqture-py/src/fermions/fermionic_hamiltonian_system.rs [hunk 2]
```diff
@@ -130,4 +131,64 @@ impl FermionHamiltonianSystemWrapper {
...
+            })?;
+            let mut fermion_system = FermionHamiltonianSystem::new(None);
+            for (key, val) in struqture_2::OperateOnDensityMatrix::iter(&two_import) {
+                let value_string = key.to_string();
+                let self_key = HermitianFermionProduct::from_str(&value_string).map_err(
+                    |_err: StruqtureError| PyValueError::new_err(
+                        "Trying to obtain struqture 1.x FermionHamiltonianSystem from struqture 2.x FermionHamiltonian. Conversion failed. Was the right type passed to all functions?".to_string()
+                ))?;
+
+                let _ = fermion_system.set(self_key, val.clone());
```
Comment (rreiner-hqs) [inline struqture-py/src/fermions/fermionic_hamiltonian_system.rs:184]: I often see this construction where we assign the result to a variable, but it seems that this way if the result of `set` is an error we won't notice, so if we don't expect an error I would handle it with `expect`.

File: struqture-py/src/fermions/fermionic_system.rs [hunk 3]
```diff
@@ -122,4 +123,62 @@ impl FermionSystemWrapper {
...
+    ///     input (FermionOperator): The struqture 2.x FermionOperator to convert to struqture 1.x.
+    ///
+    /// Returns:
+    ///     FermionSystem: The struqture 1.x FermionSystem created from the struqture 2.x FermionOperator.
+    ///
+    /// Raises:
+    ///     TypeError: If the input is not a struqture 2.x FermionOperator.
+    ///     ValueError: Conversion failed.
+    #[staticmethod]
+    pub fn from_struqture_2(input: &Bound<PyAny>) -> PyResult<FermionSystemWrapper> {
```
Comment (rreiner-hqs) [inline struqture-py/src/fermions/fermionic_system.rs:139]: Same comments about error messages and expect in this function

File: struqture-py/src/fermions/fermionic_hamiltonian_system.rs (hunk 1 above)
Comment (rreiner-hqs) [reply struqture-py/src/fermions/fermionic_hamiltonian_system.rs:150]: Also this does not sound very clear to me - maybe we can rephrase it?
```Are you sure you have the right type to all functions?```

**Review by rreiner-hqs — APPROVED** [review]

**Comment by rreiner-hqs** [comment]
Are we gonna need tests for these new functions, or are they gonna be part of compatibility crates?

File: struqture-py/src/fermions/fermionic_hamiltonian_system.rs (hunk 1 above)
Comment (kbarkhqs) [reply struqture-py/src/fermions/fermionic_hamiltonian_system.rs:150]: I have removed the second part of the error message

**Pull request merged**